packages/packages/store/src/index.ts

```diff
@@ -0,0 +1,113 @@
+import {
+	createSlice,
+	ReducersMapObject,
+	configureStore,
+	combineReducers,
+	Middleware,
+	Store,
+	Slice,
+	AnyAction,
+	CreateSliceOptions,
+} from '@reduxjs/toolkit';
+
+export type {
+	Slice,
+	CreateSliceOptions,
+	PayloadAction,
+	Store,
+	Dispatch,
+	AnyAction,
+	Action,
+	MiddlewareAPI,
+	Middleware,
+} from '@reduxjs/toolkit';
+
+export { createSlice } from '@reduxjs/toolkit';
+
+export { useSelector, useDispatch, Provider as StoreProvider } from 'react-redux';
+
+// Usage: SliceState<typeof slice>
+export type SliceState<S extends Slice> = {
+	[ key in S['name'] ]: ReturnType<S['getInitialState']>;
+}
+
+interface SlicesMap {
+	[key: Slice['name']]: Slice;
+}
+
+class StoreService {
+	private instance: Store | null = null;
+	private slices: SlicesMap = {};
+	private pendingActions: AnyAction[] = [];
+	private middlewares = new Set<Middleware>();
+
+	private getReducers() {
+		const reducers = Object.entries( this.slices ).reduce( ( reducersData: ReducersMapObject, [ name, slice ] ) => {
```

Comment:
you're reducing the reducers, does that mean the world is gonna tear apart and crash into itself?

Reply:
I hope.